src/nested_dask/core.py

```diff
@@ -102,7 +103,7 @@ def __setitem__(self, key, value):
                 new_flat = new_flat.astype({col: pd.ArrowDtype(pa.string())})
 
             # pack the modified df back into a nested column
-            meta = self._nested_meta_from_flat(new_flat, nested)
+            meta = _nested_meta_from_flat(new_flat, nested)
             packed = new_flat.map_partitions(lambda x: pack(x), meta=meta)
```

Comment:
I'd recommend to use the same dtype we have in meta with `pack(dtype=)`

Reply:
Makes sense, implemented